flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java

```diff
@@ -56,43 +56,42 @@ public static SecurityContext getInstalledContext() {
 	 * <p>Applies the configuration using the available security modules (i.e. Hadoop, JAAS).
 	 */
 	public static void install(SecurityConfiguration config) throws Exception {
+		// Install the security modules first before installing the security context
+		installModules(config);
+		installContext(config);
+	}
+
+	static void installModules(SecurityConfiguration config) throws Exception {
 
-		// install the security modules
+		// install the security module factories
 		List<SecurityModule> modules = new ArrayList<>();
 		try {
-			for (SecurityModuleFactory moduleFactory : config.getSecurityModuleFactories()) {
+			for (String moduleFactoryClass : config.getSecurityModuleFactories()) {
+				SecurityModuleFactory moduleFactory = SecurityFactoryService.findModuleFactory(moduleFactoryClass);
+				if (moduleFactory == null) {
```

Comment:
I think the service loader will throw an exception if it cannot find a factory, it will not return `null`.